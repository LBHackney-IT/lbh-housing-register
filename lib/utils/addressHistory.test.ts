import MockDate from 'mockdate';
import {
  calculateDurations,
  checkAddressHistory,
  formatDate,
} from './addressHistory';
MockDate.set('2020-06-01');

const partialEntry = {
  address: {},
  date: '',
  postcode: '',
};

describe('checkAddressHistory', () => {
  test('checkAddressHistory calculates correctly ', () => {
    expect(
      checkAddressHistory(
        [
          {
            ...partialEntry,
            date: '2019-06-01',
            dateTo: '',
          },
        ],
        1
      )
    ).toBe(true);
    expect(
      checkAddressHistory(
        [
          {
            ...partialEntry,
            date: '2020-01-01',
            dateTo: '',
          },
          {
            ...partialEntry,
            date: '2019-06-01',
            dateTo: '2020-01-01',
          },
        ],
        1
      )
    ).toBe(true);
  });
  test('checkAddressHistory only checks the last date', () => {
    expect(
      checkAddressHistory(
        [
          {
            ...partialEntry,
            date: '2019-06-01',
            dateTo: '2020-01-01',
          },
          {
            ...partialEntry,
            date: '2020-01-01',
            dateTo: '',
          },
        ],
        1
      )
    ).toBe(false);
  });
});

test('formatDate', () => {
  expect(formatDate(new Date())).toBe('June 2020');
});

test('calculateDurations calculates accurately', () => {
  expect(
    calculateDurations([
      {
        ...partialEntry,
        date: '2020-01-01',
        dateTo: '',
      },
      {
        ...partialEntry,
        date: '2019-06-01',
        dateTo: '2020-01-01',
      },
    ])
  ).toMatchSnapshot();
});
