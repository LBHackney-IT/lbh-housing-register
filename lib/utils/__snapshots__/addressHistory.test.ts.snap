// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateDurations calculates accurately 1`] = `
[
  {
    "from": 2020-01-01T00:00:00.000Z,
    "label": "0 years 5 months (January 2020 - Now)",
    "months": 5,
    "years": 0,
  },
  {
    "from": 2019-06-01T00:00:00.000Z,
    "label": "0 years 7 months (June 2019 â€“ January 2020)",
    "months": 7,
    "years": 0,
  },
]
`;
